### PR TITLE
Handle Nan in ArrayMinMax function and remove a todo.

### PR DESCRIPTION
**Pull request by amitkdutta**
Differential Revision: D49935894

### CODE REVIEW

**Comment by zacw7** [comment]
Thanks for the quick fix. Could you please double check if other similar functions also have the same issue such as `sum()`, `avg()`?

**Comment by amitkdutta** [comment]
> Thanks for the quick fix. Could you please double check if other similar functions also have the same issue such as `sum()`, `avg()`?

Will look after this one.

File: velox/docs/functions/presto/array.rst [hunk 1]
```diff
@@ -93,21 +93,26 @@ Array Functions
...
         SELECT array_max(ARRAY [-1, -2, -2]); -- -1
         SELECT array_max(ARRAY [-1, -2, NULL]); -- NULL
         SELECT array_max(ARRAY []); -- NULL
+        SELECT array_max(ARRAY [-0.0001, -0.0002, -0.0003, nan()]); -- NaN
 
 .. function:: array_min(array(E)) -> E
 
-    Returns the minimum value of input array. ::
+    Returns minimum non-NULL element of the array. Returns NULL if array is empty or contains a NULL element.
+    When E is DOUBLE or REAL, NaN value is considered greater than any non-NaN value. ::
```
Comment (bikramSingh91) [inline velox/docs/functions/presto/array.rst:108]: >Returns NULL if array is empty or contains a NULL element.
> NaN value is considered greater than any non-NaN value.

This doesn't seem to hold true completely.
NaN seems to always be returned in both min_by and max_by even if NULL exists which also means its simultaneously considered both highest and lowest value.
Tried these on presto java:
```
SELECT array_max(ARRAY [nan(), 1.0]); => NaN 
SELECT array_max(ARRAY [NULL, nan(), 1.0]); => NaN 

SELECT array_min(ARRAY [nan(), 1.0]); => NaN 
SELECT array_min(ARRAY [NULL, nan(), 1.0]); => NaN 
```

Comment (amitkdutta) [reply velox/docs/functions/presto/array.rst:108]: Good catch @bikramSingh91. In Presto, in presence of a NAN, the result is always NAN regardless of the other elements in the array for both min/max functions.

**Pull request closed without merging**